src/v2/components/database/index.ts

```diff
@@ -0,0 +1,281 @@
+      { parent: this },
+    );
+    return dbSubnetGroup;
+  }
+
+  private createSecurityGroup(
+    vpcId: awsx.ec2.Vpc['vpcId'],
+    vpcCidrBlock: pulumi.Input<string>,
+  ) {
+    const dbSecurityGroup = new aws.ec2.SecurityGroup(
+      `${this.name}-security-group`,
+      {
+        vpcId,
+        ingress: [
+          {
+            protocol: 'tcp',
+            fromPort: 5432,
+            toPort: 5432,
+            cidrBlocks: [vpcCidrBlock],
+          },
+        ],
+        tags: commonTags,
+      },
+      { parent: this },
+    );
+    return dbSecurityGroup;
+  }
+
+  private createEncryptionKey() {
+    const kms = new aws.kms.Key(
+      `${this.name}-rds-key`,
+      {
+        description: `${this.name} RDS encryption key`,
+        customerMasterKeySpec: 'SYMMETRIC_DEFAULT',
+        isEnabled: true,
+        keyUsage: 'ENCRYPT_DECRYPT',
+        multiRegion: false,
+        enableKeyRotation: true,
+        tags: commonTags,
+      },
+      { parent: this },
+    );
+    return kms;
+  }
+
+  private createMonitoringRole() {
+    const monitoringRole = new aws.iam.Role(`${this.name}-rds-monitoring`, {
+      assumeRolePolicy: {
+        Version: '2012-10-17',
+        Statement: [
+          {
+            Action: 'sts:AssumeRole',
+            Effect: 'Allow',
+            Principal: {
+              Service: 'monitoring.rds.amazonaws.com',
+            },
+          },
+        ],
+      },
+    });
+
+    new aws.iam.RolePolicyAttachment(
+      `${this.name}-rds-monitoring-role-attachment`,
+      {
+        role: monitoringRole.name,
+        policyArn:
+          'arn:aws:iam::aws:policy/service-role/AmazonRDSEnhancedMonitoringRole',
+      },
+    );
+
+    return monitoringRole;
+  }
+
+  private createEncryptedSnapshotCopy(
+    snapshotIdentifier: pulumi.Input<string>,
+  ) {
+    const sourceDbSnapshotIdentifier = pulumi
+      .output(snapshotIdentifier)
+      .apply(snapshotIdentifier =>
+        aws.rds.getSnapshot({
+          dbSnapshotIdentifier: snapshotIdentifier,
+        }),
+      )
+      .apply(snapshot => snapshot.dbSnapshotArn);
+
+    const encryptedSnapshotCopy = new aws.rds.SnapshotCopy(
+      `${this.name}-encrypted-snapshot-copy`,
+      {
+        sourceDbSnapshotIdentifier,
+        targetDbSnapshotIdentifier: `${snapshotIdentifier}-encrypted-copy`,
+        kmsKeyId: this.kmsKeyId,
+      },
+      { parent: this },
+    );
+    return encryptedSnapshotCopy;
+  }
+
+  private createParameterGroup(
+    customParameterGroupArgs: pulumi.Input<aws.rds.ParameterGroupArgs>,
+  ) {
+    const parameterGroup = pulumi
+      .output(customParameterGroupArgs)
+      .apply(args => {
+        return new aws.rds.ParameterGroup(
+          `${this.name}-parameter-group`,
+          args,
+          { parent: this },
+        );
+      });
+
+    return parameterGroup;
+  }
+
+  private createDatabaseInstance(args: Database.Args) {
+    const argsWithDefaults = Object.assign({}, defaults, args);
```

Comment:
Why is this needed? 
The `argsWithDefaults` are already created in the constructor, so calling this method from constructor with them should do the trick, e.g. `this.createDatabaseInstance(argsWithDefaults);`